ecommerce/extensions/payment/views/hyperpay.py

```diff
@@ -82,36 +135,126 @@ def _verify_status(self, resource_path):
         response_data = response.json()
 
         result_code = response_data['result']['code']
-        if not response.ok or not self.SUCCESSFUL_TRANSACTION_RESULT_CODES_REGEX.search(result_code):
-            logger.error('Received a non-success result code from HyperPay %s', result_code)
-            succeeded = False
+        if not response.ok:
+            logger.error('Received a non-success response status code from HyperPay %s', response.status_code)
+            status = PaymentStatus.FAILURE
+        elif self.PENDING_CHANGEABLE_SOON_CODES_REGEX.search(result_code):
+            logger.warning(
+                'Received a pending status code %s from HyperPay for payment id %s.',
+                result_code,
+                response_data['id']
+            )
+            status = PaymentStatus.PENDING
+        elif self.PENDING_NOT_CHANGEABLE_SOON_CODES_REGEX.search(result_code):
+            logger.warning(
+                'Received a pending status code %s from HyperPay for payment id %s. As this can change '
+                'after several days, treating it as a failure.',
+                result_code,
+                response_data['id']
+            )
+            status = PaymentStatus.FAILURE
```

Comment:
Does this status sends a warning message to sysadmins?
The transaction might get approved after a few days, no? Do we store the result code to audit this anywhere?

Reply:
All the responses are stored in the database and hence queryable. We can implement a custom cron script to check these periodically and alert the admins as it is easier to do than tweaking ecommerce to be able to do that. Ecommerce doesn't set or use `settings.ADMINS` and doesn't provide hooks/interfaces for sending emails for these states.

Reply:
Makes sense, can you schedule a follow up ticket to implement this?

Reply:
@giovannicimolin, this task is still open, and I do intend to make a lot more improvements including this one as part of the same task. Will add the details there.